core/src/main/java/hudson/model/ParametersDefinitionProperty.java

```diff
@@ -66,15 +66,22 @@
 public class ParametersDefinitionProperty extends OptionalJobProperty<Job<?, ?>>
         implements Action {
 
-    private final List<ParameterDefinition> parameterDefinitions;
+    private List<ParameterDefinition> parameterDefinitions;
 
     @DataBoundConstructor
-    public ParametersDefinitionProperty(List<ParameterDefinition> parameterDefinitions) {
-        this.parameterDefinitions = parameterDefinitions;
+    public ParametersDefinitionProperty(@CheckForNull List<ParameterDefinition> parameterDefinitions) {
+        this.parameterDefinitions = (parameterDefinitions != null) ? parameterDefinitions : Collections.<ParameterDefinition>emptyList();
     }
 
-    public ParametersDefinitionProperty(ParameterDefinition... parameterDefinitions) {
-        this.parameterDefinitions = Arrays.asList(parameterDefinitions);
+    public ParametersDefinitionProperty(@CheckForNull ParameterDefinition... parameterDefinitions) {
+        this(parameterDefinitions != null ? Arrays.asList(parameterDefinitions) : null);
+    }
+
+    protected Object readResolve() {
```

Comment:
And if we are making it private then just `return parameterDefinitions == null ? new ParametersDefinitionProperty() : this;` and the field can stay final FTW